fdb-extensions/src/test/java/com/apple/foundationdb/async/hnsw/DataRecordsTest.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * DataRecordsTest.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.async.hnsw;
+
+import com.apple.foundationdb.linear.AffineOperator;
+import com.apple.foundationdb.linear.RealVector;
+import com.apple.foundationdb.linear.RealVectorTest;
+import com.apple.foundationdb.linear.Transformed;
+import com.apple.foundationdb.tuple.Tuple;
+import com.apple.test.RandomSeedSource;
+import com.google.common.collect.ImmutableList;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.params.ParameterizedTest;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Random;
+import java.util.function.Function;
+
+class DataRecordsTest {
+    @ParameterizedTest
+    @RandomSeedSource({0x0fdbL, 0x5ca1eL, 123456L, 78910L, 1123581321345589L})
+    void testAccessInfo(final long randomSeed) {
+        assertHashCodeEqualsToString(randomSeed, DataRecordsTest::accessInfo);
+    }
+
+    @ParameterizedTest
+    @RandomSeedSource({0x0fdbL, 0x5ca1eL, 123456L, 78910L, 1123581321345589L})
+    void testAggregatedVector(final long randomSeed) {
+        assertHashCodeEqualsToString(randomSeed, DataRecordsTest::aggregatedVector);
+    }
+
+    @ParameterizedTest
+    @RandomSeedSource({0x0fdbL, 0x5ca1eL, 123456L, 78910L, 1123581321345589L})
+    void testCompactNode(final long randomSeed) {
+        final Random random = new Random(randomSeed);
+        final long dependentRandomSeed = random.nextLong();
+
+        final CompactNode compactNode1 = compactNode(new Random(dependentRandomSeed));
+        final CompactNode compactNode1Clone = compactNode(new Random(dependentRandomSeed));
+        Assertions.assertThat(compactNode1.toString()).isEqualTo(compactNode1Clone.toString());
```

Comment:
Is there a better assertion that could be used instead of asserting that the string representation is the same? I suppose that that proves that there _is_ a `toString` implementation with something more interesting than the default one, but not much else.

It also seems like it may be worth it to replace the assertion here with the suggestion from teamscale